### PR TITLE
Add alwaysPull option to solution

### PR DESCRIPTION
**Pull request by mahkoCosmo**

### CODE REVIEW

File: solution/src/main/openapi/solution.yaml [hunk 1]
```diff
@@ -749,6 +749,10 @@ components:
         repository:
           type: string
           description: the registry repository containing the image
+        alwaysPull:
+          type: boolean
+          description: the image pull policy
```
Comment (sjoubert) [inline solution/src/main/openapi/solution.yaml:754]: I guess the description is not really accurate. It doesn't actually store the pull policy.

**Review by neomatamune — CHANGES_REQUESTED** [review]
I confirm @sjoubert comments, the description of the new field would gain at being reworded (for example: "should the docker image be always pulled")
And the plantuml file should not have the `User`plus part added to it.

File: openapi/plantuml/schemas.plantuml [hunk 2]
```diff
@@ -649,8 +650,10 @@ Organization -- OrganizationServices : services
 Organization -- OrganizationSecurity : security
 FileUploadValidation -- "0..*" FileUploadMetadata : nodes
 FileUploadValidation -- "0..*" FileUploadMetadata : edges
+User -- "0..*" UserOrganization : organizations
```
Comment (sjoubert) [inline openapi/plantuml/schemas.plantuml:653]: This line should still not be there. Same for line 656 below.

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/service/ScenarioRunServiceImpl.kt [hunk 3]
```diff
@@ -497,7 +497,9 @@ class ScenarioRunServiceImpl(
     logger.debug(startInfo.toString())
     val scenarioRunRequest =
         workflowService.launchScenarioRun(
-            startInfo.startContainers, startInfo.runTemplate.executionTimeout)
+            startInfo.startContainers,
+            startInfo.runTemplate.executionTimeout,
+            startInfo.solution.alwaysPull!!)
```
Comment (sjoubert) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/service/ScenarioRunServiceImpl.kt:502]: Is `alwaysPull` nullable with a default value provided in the openapi definition? Do we need `!!`?

**Review by sjoubert — COMMENTED** [review]
Can we add a minimal test? At least to check if the constructed workflow/template has the correct policy.

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/service/ScenarioRunServiceImpl.kt (hunk 3 above)
Comment (mahkoCosmo) [reply scenariorun/src/main/kotlin/com/cosmotech/scenariorun/service/ScenarioRunServiceImpl.kt:502]: without !!, Type mismatch: inferred type is Boolean? but Boolean was expected

Comment (sjoubert) [reply scenariorun/src/main/kotlin/com/cosmotech/scenariorun/service/ScenarioRunServiceImpl.kt:502]: Ok, I admit I did not check, I just though that with a default value there was no need for the property to be nullable.

**Comment by mahkoCosmo** [comment]
> Can we add a minimal test? At least to check if the constructed workflow/template has the correct policy.

On a deja 2 tests!!!!

**Comment by sjoubert** [comment]
> > Can we add a minimal test? At least to check if the constructed workflow/template has the correct policy.
> 
> On a deja 2 tests!!!!

I see 2 tests for the `imagePullPolicy` config entry but nothing for the newly introduced `alwaysPull` property of Solution.

**Review by sjoubert — APPROVED** [review]

**Pull request merged**